### PR TITLE
Mongo userz

### PR DESCRIPTION
**Pull request by moosecanswim**
changes the mongo init config map into a secret.

to test start mesh in either k3d or minikube and make sure dashboard is up and running with all core services green.

run `kc logs data-mongo-0` to see logs of mongo.  You can also do this for the internal mongo.

### CODE REVIEW

File: data/templates/mongo-init.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  gmdata.js: |-
+  {{- with .Values.mongo.credentials }}
+    db.auth({{ quote .root_username }}, {{ quote .root_password }})
+    db = db.getSiblingDB({{ quote .database }})
+    db.createUser({
+        user: {{ quote .gmdata_username }},
+        pwd: {{ quote .gmdata_password }},
+        roles: [
+        {
+            role: {{ quote .gmdata_role }},
```
Comment (chrisbsmith) [inline data/templates/mongo-init.yaml]: @moosecanswim I updated this file.  The GM Data user does not need root rights to the admin db.  Instead. we should allow this to be configurable.  I've updated the code and given the `readWrite` default.

File: data/values.yaml [hunk 2]
```diff
@@ -250,8 +250,9 @@ mongo:
     root_username: 'mongo'
     root_password: 'mongo'
     database: 'gmdata'
-    gmdata_username: 'gmdata-user'
-    gmdata_password: 'mongopassword'
+    gmdata_username: 'gmdata'
```
Comment (chrisbsmith) [inline data/values.yaml]: We should also not use the same username and password for the GM Data user as the root user, b/c we'll mask any potential permissions issues.

**Comment by moosecanswim** [comment]
Can do

**Comment by moosecanswim** [comment]
Works as expected and i incremented the chart version

**Pull request merged**